tremor-script/src/ast/query/raw.rs

```diff
@@ -17,30 +17,73 @@
 
 use super::super::raw::{ExprRaw, IdentRaw, ImutExprRaw, ModuleRaw, ScriptRaw, WithExprsRaw};
 use super::{
+    super::{ReservedPath, Segment},
     error_generic, error_no_consts, error_no_locals, AggrRegistry, BaseExpr, GroupBy, GroupByInt,
-    HashMap, Helper, ImutExpr, Location, NodeMetas, OperatorDecl, OperatorKind, OperatorStmt,
-    Query, Registry, Result, ScriptDecl, ScriptStmt, Select, SelectStmt, Serialize, Stmt,
-    StreamStmt, Upable, Value, WindowDecl, WindowKind,
+    HashMap, Helper, ImutExpr, ImutExprInt, Location, NodeMetas, OperatorDecl, OperatorKind,
+    OperatorStmt, Path, Query, Registry, Result, ScriptDecl, ScriptStmt, Select, SelectStmt,
+    Serialize, Stmt, StreamStmt, SubqueryDecl, Upable, Value, WindowDecl, WindowKind,
 };
 use crate::ast::visitors::{GroupByExprExtractor, TargetEventRefVisitor};
 use crate::{ast::InvokeAggrFn, impl_expr};
 use beef::Cow;
 
 fn up_params<'script, 'registry>(
-    params: WithExprsRaw<'script>,
+    params: Params<'script>,
     helper: &mut Helper<'script, 'registry>,
 ) -> Result<HashMap<String, Value<'script>>> {
-    params
-        .into_iter()
-        .map(|(name, value)| Ok((name.to_string(), value.up(helper)?.try_into_value(helper)?)))
-        .collect()
+    match params {
+        Params::Raw(params) => params
+            .into_iter()
+            .map(|(name, value)| Ok((name.to_string(), value.up(helper)?.try_into_value(helper)?)))
+            .collect(),
+        Params::Processed(params) => Ok(params),
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
+/// we're forced to make this pub because of lalrpop
+pub enum Params<'script> {
+    Raw(WithExprsRaw<'script>),
+    Processed(HashMap<String, Value<'script>>),
+}
+impl<'script> From<WithExprsRaw<'script>> for Params<'script> {
+    fn from(raw: WithExprsRaw<'script>) -> Self {
+        Params::Raw(raw)
+    }
+}
+impl<'script> From<HashMap<String, Value<'script>>> for Params<'script> {
+    fn from(processed: HashMap<String, Value<'script>>) -> Self {
+        Params::Processed(processed)
+    }
 }
 
 fn up_maybe_params<'script, 'registry>(
-    params: Option<WithExprsRaw<'script>>,
+    params: MaybeParams<'script>,
     helper: &mut Helper<'script, 'registry>,
 ) -> Result<Option<HashMap<String, Value<'script>>>> {
-    params.map(|params| up_params(params, helper)).transpose()
+    match params {
+        MaybeParams::Raw(params) => params
+            .map(|params| up_params(params.into(), helper))
+            .transpose(),
+        MaybeParams::Processed(params) => Ok(params),
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize)]
```

Comment:
this would be better solved by `Option<Params<'script>>` that way there is no need to duplicate the code